provider/core/src/marker.rs

```diff
@@ -84,6 +84,15 @@ pub trait DataMarker: 'static {
 pub trait KeyedDataMarker: DataMarker {
     /// The single [`DataKey`] associated with this marker.
     const KEY: DataKey;
+
+    /// Binds this [`KeyedDataMarker`] to a provider supporting it.
+    fn bind<P>(provider: P) -> DataProviderWithKey<Self, P>
```

Comment:
I added this helper function because type inference works better at the call site. Instead of doing

```rust
DataProviderWithKey::<MyDataMarkerV1, _>::new(provider)
```

I can just write

```rust
MyDataMarkerV1::bind(provider)
```